packages/bvaughn-architecture-demo/src/contexts.ts

```diff
@@ -0,0 +1,48 @@
+import { ExecutionPoint } from "@replayio/protocol";
+import { createContext } from "react";
+
+import { Range } from "./types";
+
+export type ConsoleLevelFlags = {
+  showErrors: boolean;
+  showLogs: boolean;
+  showWarnings: boolean;
+};
+
+export type ConsoleFiltersContextType = {
+  // Filter text to display in the UI.
+  // This value is updated at React's default, higher priority.
+  filterByDisplayText: string;
+
+  // Text to filter console messages by.
+  // This value is updated at a lower, transition priority.
+  filterByText: string;
+
+  // Filter by text is about to be updated as part of a transition;
+  // UI that consumes the focus for Suspense purposes may wish want reflect the temporary pending state.
+  isTransitionPending: boolean;
+
+  // Types of console messages to include (or filter out).
+  levelFlags: ConsoleLevelFlags;
+
+  update: (filterByText: string, levelFlags: ConsoleLevelFlags) => void;
+};
```

Comment:
These contexts are nice for sharing state between different parts of the app (without "props drilling"). I guess they also give us a nice _super lightweight_ way to statically render our UI if we want to for visual testing (but I haven't thought about this use case much yet).